src/Sulu/Component/Content/Document/Subscriber/WebspaceSubscriber.php

```diff
@@ -49,13 +62,39 @@ public function supports($document)
     }
 
     /**
-     * @param HydrateEvent $event
+     * @param AbstractMappingEvent|HydrateEvent $event
+     * @throws \Sulu\Component\DocumentManager\Exception\DocumentManagerException
      */
     public function doHydrate(AbstractMappingEvent $event)
     {
         $document = $event->getDocument();
-        $webspace = $this->inspector->getWebspace($document);
-        $event->getAccessor()->set('webspaceName', $webspace);
+        $webspaceName = $this->inspector->getWebspace($document);
+        $event->getAccessor()->set('webspaceName', $webspaceName);
+
+        if ($document instanceof StructureBehavior && $document->getStructureType() === null) {
+            $this->setDefaultStructureType($event, $document, $webspaceName);
+        }
```

Comment:
well, we can do that once: https://github.com/sulu-io/sulu/pull/1272 is merged. Or, we can just throw an exception only when the structure data is not empty - so if there is structure data, and no structure type has been provided.